recipe/build.sh

```diff
@@ -1,5 +1,23 @@
 #!/bin/bash
+
+if [ -z "$LD_LIBRARY_PATH" ]; then
+    old_ld_library_path="$LD_LIBRARY_PATH"
+fi
+
+export LD_LIBRARY_PATH=$(pwd)
+
+# world-writable files are not allowed
+chmod -R o-w $SRC_DIR
+
 sh Configure -de -Dprefix=$PREFIX -Duserelocatableinc
 make
+
+# change permissions again after building
+chmod -R o-w $SRC_DIR
```

Comment:
Built files may have incorrect permissions, so the script corrects before building and again after. Tests were previously failing because various files were world-writable.